packages/core/graph/src/AdjacencyList.js

```diff
@@ -0,0 +1,1195 @@
+// @flow
+import assert from 'assert';
+import {fromNodeId, toNodeId} from './types';
+import type {NullEdgeType, AllEdgeTypes} from './Graph';
+import type {NodeId} from './types';
+
+/**
```

Comment:
The implementation described here was inspired by [This dissection of the v8 Map internals](https://itnext.io/v8-deep-dives-understanding-map-internals-45eb94a183df)